src/test/kotlin/se/uulm/snowballr/backend/service/paper/GetForwardReferencedPapersTest.kt

```
@@ -0,0 +1,131 @@
+                citationRepoMock.getForwardReferencedPaperIdsOfPaperById(paper.id)
+            } throws TestSpecificException()
+            return
+        }
+        coEvery {
+            citationRepoMock.getForwardReferencedPaperIdsOfPaperById(paper.id)
+        } returns listOf(forwardReferenceId)
+
+        if (failAt == paperRepoMock::getPaperById) {
+            coEvery { paperRepoMock.getPaperById(forwardReferenceId) } throws TestSpecificException()
+            return
+        }
+        coEvery { paperRepoMock.getPaperById(forwardReferenceId) } returns citingPaper
+
+        if (failAt == authorOfPaperRepoMock::getAuthorsOfPaperById) {
+            coEvery { authorOfPaperRepoMock.getAuthorsOfPaperById(forwardReferenceId) } throws TestSpecificException()
+            return
+        }
+        coEvery { authorOfPaperRepoMock.getAuthorsOfPaperById(forwardReferenceId) } returns listOf(author)
+
+        if (failAt == citationRepoMock::getBackwardsReferencedPaperIdsOfPaperById) {
+            coEvery {
+                citationRepoMock.getBackwardsReferencedPaperIdsOfPaperById(forwardReferenceId)
+            } throws TestSpecificException()
+            return
+        }
+        coEvery {
+            citationRepoMock.getBackwardsReferencedPaperIdsOfPaperById(forwardReferenceId)
+        } returns listOf(UUID.randomUUID())
+    }
+
+    @ParameterizedTest
+    @MethodSource("failingFunctions")
+    fun `When a step fails, then an exception is thrown`(failAt: KFunction<*>) = runTest {
+        mockHappyPathUntil(failAt)
+        assertThrows<TestSpecificException> {
+            mainService.getForwardReferencedPapers(getExampleRequest())
+        }
+    }
+
+    @Test
+    fun `When the paper doesn't exist, then a NotFoundException is thrown`() = runTest {
+        coEvery { paperRepoMock.doesPaperExistById(requestId) } throws SnowballRException.NotFoundException(
+            entityType = EntityType.PAPER, requestId.toString(),
+        )
```

Here we want to test what happens when the paper does not exist, but instead of returning `false` for `doesPaperExistById` we mock that this method throws an exception, which is already tested in the happy path testing before.